### DISCUSSION TITLE
Can someone confirm this bug: Download of (stacked) images not working in Build 220629-5d7448d2

### DISCUSSION THREAD

**Discussion by dinoschachten** [photoprism/photoprism#2542] [Q&A] [answered]
#### 1. What is not working as documented?

After updating to Build 220629-5d7448d2 downloading of stacked images no longer works (tested in Chrome 103 and Safari 13 on macOS). I observed this with RAW files (Sony ARW) that had JPG versions created automatically by Photoprism. JPG originals (like smartphone photos) still download as expected.

#### 2. How can we reproduce it?

1. Configure Photoprism to automatically create JPG versions of RAW files
2. Index a RAW file so that a stack is created
3. In the Search or Album view, select a RAW file
4. In the actions menu in the bottom right corner, click on download (the same can be observed when opening the image viewer for a stacked photo and clicking on the download button in the top right corner)
5. A progress bar will appear briefly, but no further response from Photoprism or the browser will occur, no download will occur

On [demo.photoprism.app](https://demo.photoprism.app/browse) I was not able to fully reproduce this: There are some RAW files there, when I click the download button for one of them, only JPGs are being downloaded. Not sure if this is due to configuration or a different flavour of the issue I am having.

#### 3. What behaviour do you expect?

Prior to the update, both the original RAW file as well as the generated JPG version would be downloaded when clicking on the download button. I have not changed any configuration, I have tested enabling and disabling experimental features afterwards, both options have no influence on the behaviour.

#### 4. What could be the cause of your problem?

I don't feel qualified to answer that :) However, I do see a JS error in Chrome's inspector - see below

#### 5. Can you provide us with example files for testing, error logs, or screenshots?

Uploaded a screenshot of the JS error here: https://cloud.dreamwood-music.com/nextcloud/index.php/s/cyYH9rD343xeMc5

Curiously, despite activating the extended debug logs and restarting, nothing shows up in the WebUI logs.

#### 6. Which software versions do you use?

- PhotoPrism Plus Build 220629-5d7448d2
- macOS 10.13, macOS 10.15
- Chrome 103, Safari 13

#### 7. On what kind of device is PhotoPrism installed?

Ubuntu 20.04 LTS, workstation PC, local network.

#### 8. Do you use a Reverse Proxy, Firewall, VPN, or CDN?

No.

**Comment by graciousgrey** [comment, marked as answer]
A few versions ago we introduced some download settings.

The default download settings for downloads from the context menu are:
Originals: true
MediaRaw: false
MediaSidecar: false

Originals:true means only files from your originals folder will be downloaded, not from sidecar.
Downloading from the files tab of the edit dialog is not affected by these settings.

You find these settings in the settings.yml which is located in storage/config. These settings will be added to the UI in a later version.

**In case you want to download RAW and JPG change it to:**
Originals: false
MediaRaw: true
MediaSidecar: false

**In case you want to download only the RAW  change it to:**
Originals: true
MediaRaw: true
MediaSidecar: false

**In case you want to download only the JPG change it to:**
Originals: false
MediaRaw: false
MediaSidecar: false

Don't forget to stop and start PhotoPrism for changes to be applied.

I hope this helps :)

**Reply by dinoschachten** [reply]
Yes - awesome, thank you! Works like a charm!

I really hadn't considered that there are more download settings outside the UI. I admit I was not aware of the settings.yml (only the docker-compose.yml).
I believe in the [docs](https://docs.photoprism.app/user-guide/settings/advanced/) the settings.yaml isn't being mentioned by name and as a place to make additional settings not available in the UI yet. Perhaps I could and should make a contribution in that regard some time.

**Reply by graciousgrey** [reply]
It's not in the docs intentionally as all those options should be available in the UI it's just a matter of time :)
